tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -803,3 +805,361 @@ def test_csrlsvqr(self, dtype):
                                    reorder=self.reorder)
         cupy.testing.assert_allclose(x, ref_x, rtol=test_tol,
                                      atol=test_tol)
+
+
+@testing.with_requires('scipy>=1.4')
+@testing.gpu
+# tests adapted from scipy's tests of lobpcg
+class TestLOBPCG:
+
+    def _elastic_rod(self, n, xp):
+        """Build the matrices for the generalized eigenvalue problem of the
+        fixed-free elastic rod vibration model.
+        """
+        L = 1.0
+        le = L / n
+        rho = 7.85e3
+        S = 1.e-4
+        E = 2.1e11
+        mass = rho * S * le / 6.
+        k = E * S / le
+        A = k * (xp.diag(xp.r_[2. * xp.ones(n - 1), 1]) -
+                 xp.diag(xp.ones(n - 1), 1) - xp.diag(xp.ones(n - 1), -1))
+        B = mass * (xp.diag(xp.r_[4. * xp.ones(n - 1), 2]) +
+                    xp.diag(xp.ones(n - 1), 1) + xp.diag(xp.ones(n - 1), -1))
+        return A, B
+
+    def _mikota_pair(self, n, xp):
+        """Build a pair of full diagonal matrices for the generalized eigenvalue
+        problem. The Mikota pair acts as a nice test since the eigenvalues are
+        the squares of the integers n, n=1,2,...
+        """
+        x = xp.arange(1, n + 1)
+        B = xp.diag(1. / x)
+        y = xp.arange(n - 1, 0, -1)
+        z = xp.arange(2 * n - 1, 0, -2)
+        A = xp.diag(z) - xp.diag(y, -1) - xp.diag(y, 1)
+        return A, B
+
+    def _compare_solutions(self, A, B, m, xp, sp):
+        """Check eig vs. lobpcg consistency.
+        """
+        n = A.shape[0]
+        V = testing.shaped_random((n, m), xp=numpy)
+        X = scipy.linalg.orth(V)
+        eigvals, eigvecs = sp.linalg.lobpcg(A,
+                                            xp.asarray(X), B=B,
+                                            tol=1e-5, maxiter=30,
+                                            largest=False)
+        return eigvals, xp.absolute(eigvecs)
```

Comment:
`xp.absolute` does not always return a vector parallel to the original vector.

Reply:
I have used a strategy under `_eigen_vec_transform` that standardizes the eigen vectors based on their first non-zero entries. LMK if that works